addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/model/AggregateDataResponse.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solaredge.internal.model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * this class is used to map the aggregate data json response
+ *
+ * @author Alexander Friese - initial contribution
+ *
+ */
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AggregateDataResponse implements DataResponse {
+
+    private static final String UNIT_WH = "Wh";
+    private static final String UNIT_KWH = "KWh";
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class Value {
+        public String value;
+        public String unit;
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class ValueAndPercent extends Value {
+        public String percentage;
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class UtilizationMeasures {
+        public Value production;
+        public Value consumption;
+        public ValueAndPercent selfConsumptionForConsumption;
+        public ValueAndPercent batterySelfConsumption;
+        @JsonProperty("import")
+        public ValueAndPercent imported;
+        public ValueAndPercent export;
+    }
+
+    private UtilizationMeasures utilizationMeasures;
+
+    // TODO: use units!
```

Comment:
What do you mean with this TODO. Maybe clarify or remove?

Reply:
this has already been implemented. In the first version all values where expected to be with unit KWh but actually it could also be Wh. Therefore the "unit" which is provided in the json must be taken into account and all values need to be converted to KWh to stay consistent.
This is now donw by method "getValueAsKWh(Value value)"